### PR TITLE
DP-26263: Do not add or change JS org variable tracking when on a top…

### PR DESCRIPTION
**Pull request by arthurbaghdas**
…ic page

**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-26263

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_content/mass_content.module [hunk 1]
```diff
@@ -537,7 +537,13 @@ function mass_content_theme() {
  */
 function mass_content_page_attachments(array &$attachments) {
   $attachments['#attached']['library'][] = 'mass_content/contextual-nav';
-  $attachments['#attached']['library'][] = 'mass_content/session_organizations';
+  $route_name = \Drupal::routeMatch()->getRouteName();
+  if ($route_name === 'entity.node.canonical') {
+    $entity = \Drupal::routeMatch()->getParameter('node');
+    if ($entity && $entity->bundle() !== 'topic_page') {
+      $attachments['#attached']['library'][] = 'mass_content/session_organizations';
```
Comment (weitzman) [inline docroot/modules/custom/mass_content/mass_content.module]: I think you also have to add cacheability metadata or else the dynamic page cache for a topic_page could be used on a non-topic page and vice versa. See https://www.drupal.org/project/drupal/issues/2478393 for an example patch which adds this data.

Comment (arthurbaghdas) [reply docroot/modules/custom/mass_content/mass_content.module]: I've added cache context to the route entity.node.canonical.

Comment (weitzman) [reply docroot/modules/custom/mass_content/mass_content.module]: I'm not sure thats right. That context means that we will have different caches for node-canonical pages and non-node-canonical. But the code attaches the library based on topic page versus non-topic page. So node-canonical is not the right bucket.

However, I dont see a per-bundle cache context in drupal core. So maybe I am missing something. Can we always attach mass_content/session_organizations library but have the javascript not take action for topic pages? Javascript has access to entity type via Drupal.settings object, I think.

Comment (arthurbaghdas) [reply docroot/modules/custom/mass_content/mass_content.module]: Refactored the code according to the request.

**Review by weitzman — APPROVED** [review]

**Pull request merged**